app/src/main/kotlin/com/wire/android/util/DateAndTimeParsers.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+@file:Suppress("TooGenericExceptionCaught")
+
+package com.wire.android.util
+
+import androidx.compose.runtime.Stable
+import com.wire.android.appLogger
+import kotlinx.datetime.Instant
+import kotlinx.datetime.toJavaInstant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+import java.util.Date
+import java.util.Locale
+
+//region convenience ext functions, for retro compatibility
+@Stable
+fun String.serverDate(): Date? = DateAndTimeParsers.serverDate(this)
+
+@Stable
+fun String.deviceDateTimeFormat(): String? = DateAndTimeParsers.deviceDateTimeFormat(this)
+
+@Stable
+fun String.formatMediumDateTime(): String? = DateAndTimeParsers.formatMediumDateTime(this)
+
+@Stable
+fun String.formatFullDateShortTime(): String? = DateAndTimeParsers.formatFullDateShortTime(this)
```

Comment:
Same here. It would be nice to deprecate these.

Reply:
Thanks for the comments, I think deprecating this was not part of the ticket, since we are focusing on removing the older APIs 🤔  and focusing in perf of SimpleDateFormat vs DateTimeFormatter (from java.time API)

Reply:
I think we can address this in another ticket/effort, unless you see another option.

Reply:
After syncing, I Added anyway the deprecation warnings for each of one related to `strings`.